### PR TITLE
Add element state to access

### PR DESCRIPTION
**Pull request by owickstrom**
Right now only one `ElementState` per access, but should probably be a set, preferable that constructs an object.

### CODE REVIEW

File: lib/Specstrom/Evaluator.hs [hunk 1]
```diff
@@ -109,7 +116,11 @@ initialEnv =
       ("always_", PartialOp AlwaysOp []),
       ("eventually_", PartialOp EventuallyOp []),
       ("next_", PartialOp NextOp []),
-      ("access", PartialOp MkAccessor [])
+      ("access", PartialOp MkAccessor []),
+      ("text", Independent $ LitVal (ElementStateLit Text)),
+      ("cssValue", PartialOp MkCssValue []),
+      ("attribute", PartialOp MkAttribute []),
+      ("property", PartialOp MkProperty [])
```
Comment (liamoc) [inline lib/Specstrom/Evaluator.hs:123]: If these are literals, which they appear to be, they shouldn't be in the list of built-in ops. You should generate them directly from the parser instead.

File: lib/Specstrom/Evaluator.hs [hunk 2]
```diff
@@ -159,10 +170,11 @@ app (PartialOp o [v1]) v2
   | o `elem` [Less, LessEq, Greater, GreaterEq] = binaryComparisonOp o v1 v2
   | o `elem` [AndOp, OrOp, ImpliesOp] = binaryBooleanOp o v1 v2
   | o `elem` [UntilOp] = binaryTemporalOp o v1 v2
+  | o `elem` [MkAccessor] = binarySelectorOp o v1 v2
 app (PartialOp o []) v
   | o `elem` [NotOp] = unaryBooleanOp o v
   | o `elem` [AlwaysOp, EventuallyOp, NextOp] = unaryTemporalOp o v
-  | o `elem` [MkAccessor] = unarySelectorOp o v
+  | o `elem` [MkCssValue, MkProperty, MkAttribute] = unaryElementStateOp o v
```
Comment (liamoc) [inline lib/Specstrom/Evaluator.hs:177]: These ops should not exist

File: lib/Specstrom/Evaluator.hs [hunk 3]
```diff
@@ -295,6 +307,16 @@ unaryTemporalOp o (Formula a f) = Right $ Formula a (funFor o f)
...
+unaryElementStateOp :: Op -> Value -> Either EvalError Value
+unaryElementStateOp o (Independent (LitVal (StringLit name))) =
+  let es = case o of
+        MkCssValue -> CssValue name
+        MkProperty -> Property name
+        MkAttribute -> Attribute name
+   in Right (Independent (LitVal (ElementStateLit es)))
+
+binarySelectorOp :: Op -> Value -> Value -> Either EvalError Value
+binarySelectorOp _o (Independent (LitVal (SelectorLit a))) (Independent (LitVal (ElementStateLit state))) =
```
Comment (liamoc) [inline lib/Specstrom/Evaluator.hs:319]: We should bikeshed the `access selector state` syntax. Maybe an infix operator?

File: test.strom [hunk 4]
```diff
@@ -11,11 +11,14 @@ syntax always_ 6 right;
 syntax next_ 6 right;
 syntax eventually_ 6 right;
 syntax _until_ 5 right;
+syntax css_ 1;
```
Comment (liamoc) [inline test.strom:14]: This is not defined anywhere?

File: lib/Specstrom/Evaluator.hs (hunk 1 above)
Comment (liamoc) [reply lib/Specstrom/Evaluator.hs:123]: Although it would make quickLTL more Web-specific to bake it in the parser, I don't think that matters for now

Comment (owickstrom) [reply lib/Specstrom/Evaluator.hs:123]: Hmm, yeah maybe that's better to ensure they're literals (and thus statically known). I'll change it, might need some help to figure out how.

**Pull request closed without merging**